MediaBrowser.Controller/IO/FileSystemHelper.cs

```diff
@@ -61,4 +61,77 @@ public static void DeleteEmptyFolders(IFileSystem fileSystem, string path, ILogg
             }
         }
     }
+
+    /// <summary>
+    /// Gets the target of the specified file link.
+    /// </summary>
+    /// <remarks>
+    /// This helper exists because of this upstream runtime issue; https://github.com/dotnet/runtime/issues/92128.
+    ///
+    /// It is adapted to suit the use case of the consumers of this helper.
+    /// </remarks>
+    /// <param name="linkPath">The path of the file link.</param>
+    /// <param name="returnFinalTarget">true to follow links to the final target; false to return the immediate next link.</param>
+    /// <returns>
+    /// A <see cref="FileInfo"/> if the <paramref name="linkPath"/> is a link, regardless of if the target exists; otherwise, <c>null</c>.
+    /// </returns>
+    public static FileInfo? ResolveLinkTarget(string linkPath, bool returnFinalTarget = false)
+    {
+        // Check if the file exists so the native resolve handler won't throw at us.
+        if (!File.Exists(linkPath))
+        {
+            return null;
+        }
+
+        var targetInfo = (FileInfo?)null;
+        if (returnFinalTarget)
+        {
+            // We're running the loop ourselves because of the upstream runtime issue (see remarks section of method).
+            // Because of this, we can't incur a nested IOException, but we do need to look out for infinite loops.
```

Comment:
I don't think we need these remarks, should be obvious from the mention in the method description and the comments within the loop